src/app.ts

```diff
@@ -12,15 +12,31 @@ interface TemperatureSummary {
   low: number
   average: number
 }
+let data: TemperatureReading[] = []
 
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  data = [...readings]
 }
 
 export function getTemperatureSummary(
   date: Date,
   city: string,
 ): TemperatureSummary | null {
-  //add here your code
-  return null
+  const temperatures = [] as number[]
+  let avg = 0
+  data.forEach((item) => {
+    if (item.city === city && item.time.getTime() === item.time.getTime()) {
+      temperatures.push(item.temperature)
+      avg += item.temperature
+    }
+  })
+  const temperaturesLength = temperatures.length
+  if (!temperaturesLength) return null
+  return {
+    first: temperatures[0],
+    last: temperatures[temperaturesLength - 1],
+    high: Math.max(...temperatures),
+    low: Math.min(...temperatures),
```

Comment:
Internally, Math.max and Math.max will apply a sort in order to get the maximum and minimum values, so you could have leveraged using by yourself a sort and then get the first and the last value result[0] and result[result.length - 1]